JotunnLib/Managers/DungeonManager.cs

```diff
@@ -0,0 +1,372 @@
+
+            return true;
+        }
+
+
+        /// <summary>
+        ///     Adds a new environment prefab to be registered when <see cref="ZoneSystem.SetupLocations"/> runs.<br />
+        ///     If you intend to use a custom interior environment <see cref="Location.m_interiorEnvironment"/>, this method enables you 
+        ///     to provide a prefab with an appropriately configured <see cref="LocationList"/> containing atleast one
+        ///     <see cref="EnvSetup"/> within <see cref="LocationList.m_environments"/>.
+        /// </summary>
+        /// <param name="assetBundle">The <see cref="AssetBundle"/> containing the prefab.</param>
+        /// <param name="prefabName">The name of the prefab to register.</param>
+        public void RegisterEnvironment(AssetBundle assetBundle, string prefabName)
+        {
+            if (assetBundle == null)
+            {
+                throw new ArgumentException("Cannot be null", nameof(assetBundle));
+            }
+            if (string.IsNullOrEmpty(prefabName))
+            {
+                throw new ArgumentException("Cannot be null or empty", nameof(prefabName));
+            }
+            var env = assetBundle.LoadAsset<GameObject>(prefabName);
+            loadedEnvironments.Add(prefabName, env);
+        }
+
+        private void GenerateHashes()
+        {
+            hashToName.Clear();
+
+            foreach (CustomRoom room in Rooms.Values)
+            {
+                int stableHashCode = room.Prefab.name.GetStableHashCode();
+                if (hashToName.ContainsKey(stableHashCode))
+                {
+                    Logger.LogWarning($"Room {room.Name} is already registered with hash {stableHashCode}");
+                }
+                else
+                {
+                    hashToName.Add(stableHashCode, room.Name);
+                }
+            }
+        }
+
+        private void OnZoneSystemSetupLocations()
+        {
+            // TODO - unsure if cache clearing is still necessary for resolving mocks of these assets
+            //ClearPrefabCache(typeof(Material));
+
+            if (loadedEnvironments.Count > 0)
+            {
+                foreach (var kvp in loadedEnvironments)
+                {
+                    Logger.LogDebug($"Registering environment {kvp.Key}.");
+
+                    var env = kvp.Value;
+                    env.FixReferences(true);
+                    UnityEngine.Object.Instantiate<GameObject>(env, DungeonRoomContainer.transform);
+                }
+            }
+
+            if (prefabNames.Count > 0)
+            {
+                foreach (var kvp in prefabNames)
+                {
+                    Logger.LogDebug($"Registering prefab {kvp.Key}.");
+
+                    var env = kvp.Value.LoadAsset<GameObject>(kvp.Key);
+                    env.FixReferences(true);
+                    PrefabManager.Instance.AddPrefab(env);
+                }
+            }
```

Comment:
Obsolete code that should be removed, together with the prefabNames declaration above

Reply:
Addressed in https://github.com/Valheim-Modding/Jotunn/pull/434/commits/19fad19f47d473bfc5f4e86696359412a77e02b3